bigquery_etl/backfill/shredder_mitigation.py

```diff
@@ -50,30 +67,187 @@ class DataTypeGroup(Enum):
+    name: str
+    data_type: DataTypeGroup = attr.field(default=DataTypeGroup.UNDETERMINED)
+    column_type: ColumnType = attr.field(default=ColumnType.UNDETERMINED)
+    status: ColumnStatus = attr.field(default=ColumnStatus.UNDETERMINED)
+
+    """Validate the type of the attributes."""
+
+    @data_type.validator
+    def validate_data_type(self, attribute, value):
+        """Check that the type of data_type is as expected."""
+        if not isinstance(value, DataTypeGroup):
+            raise ValueError(f"Invalid {value} with type: {type(value)}.")
+
+    @column_type.validator
+    def validate_column_type(self, attribute, value):
+        """Check that the type of parameter column_type is as expected."""
+        if not isinstance(value, ColumnType):
+            raise ValueError(f"Invalid data type for: {value}.")
+
+    @status.validator
+    def validate_status(self, attribute, value):
+        """Check that the type of parameter column_status is as expected."""
+        if not isinstance(value, ColumnStatus):
+            raise ValueError(f"Invalid data type for: {value}.")
+
+
+@attr.define
+class Subset:
+    client: bigquery.Client()
+    destination_table: str = attr.field()
+    suffix: str = attr.field(converter=attr.converters.default_if_none(""))
+    query_cte: str = attr.field(converter=attr.converters.default_if_none(""))
+    dataset: str = attr.field(default=TEMP_DATASET)
+    project_id: str = attr.field(default=DEFAULT_PROJECT_ID)
+    expiration_days: Optional[float] = attr.field(default=None)
+
+    @destination_table.validator
+    def validate_table_version(self, attribute, value):
+        """Raise an exception if the version table name doesn't end with an integer."""
+        match = re.search(r"v(\d+)$", self.destination_table)
+        try:
+            version = int(match.group(1))
+            if not isinstance(version, int) or version <= 0:
+                raise click.ClickException(
+                    f"Destination table {value} must end with a positive version number."
+                )
+        except AttributeError:
+            f"Invalid table version for {value}"
+
+    @property
+    def expiration_ms(self) -> Optional[float]:
+        """Convert partition expiration from days to milliseconds."""
+        if self.expiration_days is None:
+            return None
+        return int(self.expiration_days * 86400000)
+
+    @property
+    def name(self):
+        if self.suffix is not None and self.suffix != "":
+            return f"{self.destination_table}{self.suffix}"
+        else:
+            return self.destination_table
+
+    @property
+    def version(self):
+        match = re.search(r"v(\d+)$", self.destination_table)
+        return int(match.group(1))
+
+    @property
+    def name_previous_version(self):
+        """Return the table name for the previous version of the table."""
+        return f"{self.destination_table[:-1]}{int(self.version)-1}"
```

Comment:
oh and also ... if there is a table `_v10` only removing the last digit would return `v19` 
maybe use the parsed version to remove the right amount of strings 
```suggestion
        return f"{self.destination_table.removesuffix(str(self.version)}{self.version-1}"
```
also .. you don't need to `int()` the version as it is returned as an integer

Reply:
I've added a validation in the function `if new.version < 2:`

Reply:
Thanks for catching the problem removing only the last digit! I've moved this outside the class because it's not common to all subsets and I'm using the version for this. It's now called `destination_table_previous_version`.